chrome/content/zotero/xpcom/data/feedItems.js

```diff
@@ -94,6 +94,35 @@ Zotero.FeedItems = new Proxy(function() {
 		return this.getAsync(id);
 	});
 	
+	this.toggleReadByID = Zotero.Promise.coroutine(function* (ids, state) {
+		if (!Array.isArray(ids)) {
+			if (typeof ids != 'string') throw new Error('ids must be a string or array in Zotero.FeedItems.toggleReadByID');
+			
+			ids = [ids];
+		}
+		let items = yield this.getAsync(ids);
+		
+		if (state == undefined) {
+			// If state undefined, toggle read if at least one unread
+			state = false;
+			for (let item of items) {
+				if (!item.isRead) {
+					state = true;
+					break;
+				}
+			}
+		}
+		
+		for (let i=0; i<items.length; i++) {
+			items[i].toggleRead(state);
```

Comment:
That's only for very top-level code to decide, though (e.g., event handlers). Data layer functions should finish their work before returning. (If they don't, we end up with race conditions in tests, etc.)